### PR TITLE
ESP32: NimBLE - Fix issues during secure pairing over BLE

### PR DESCRIPTION
**Pull request by dhrishi**
#### Problem
After Rendezvous is complete and BLE link is diconnected, the device cannot start BLE advertising due to incorrect state

 #### Summary of Changes
1. Propagate the correct state DriveBLEState
2. Return checks for ESP32 NimBLE APIs should not use CHIP error codes directly
3. Make the logs during the secure pairing more clearer

With this change, tested that Rendezvous BLE with NimBLE host is working fine with `examples/chiptool`

### CODE REVIEW

**Review by saurabhst — APPROVED** [review]

File: src/platform/ESP32/nimble/BLEManagerImpl.cpp [hunk 1]
```diff
@@ -1040,33 +1040,32 @@ CHIP_ERROR BLEManagerImpl::StartAdvertising(void)
...
         }
         else
         {
-            err = ble_gap_adv_start(own_addr_type, NULL, BLE_HS_FOREVER, &adv_params, ble_svr_gap_event, NULL);
-            if (err != CHIP_NO_ERROR)
-            {
-                ChipLogError(DeviceLayer, "Retried adv start; ble_gap_adv_start() failed: %s", ErrorStr(err));
-            }
+            ChipLogError(DeviceLayer, "ble_gap_adv_start() failed: %s", ErrorStr(err));
+            return CHIP_ERROR_INTERNAL;
```
Comment (dhrishi) [inline src/platform/ESP32/nimble/BLEManagerImpl.cpp:1066]: Returning `CHIP_ERROR_INTERNAL` as of now. The errors should ideally be mapped to CHIP error codes. Filed an issue here: https://github.com/project-chip/connectedhomeip/issues/4256

File: src/platform/ESP32/nimble/BLEManagerImpl.cpp [hunk 2]
```diff
@@ -1040,33 +1040,32 @@ CHIP_ERROR BLEManagerImpl::StartAdvertising(void)
...
+            }
+            else
+            {
+                err = ble_gap_adv_start(own_addr_type, NULL, BLE_HS_FOREVER, &adv_params, ble_svr_gap_event, NULL);
+            }
         }
-    }
 
-exit:
-    if (err == BLE_HS_EALREADY && connectable)
```
Comment (bzbarsky-apple) [inline src/platform/ESP32/nimble/BLEManagerImpl.cpp:1051]: So why was this not working?  Is `connectable` false in the cases when it failed?

File: src/platform/ESP32/nimble/BLEManagerImpl.cpp [hunk 3]
```diff
@@ -470,7 +470,7 @@ void BLEManagerImpl::DriveBLEState(void)
         if (GetFlag(mFlags, kFlag_Advertising))
         {
             err = ble_gap_adv_stop();
-            if (err != CHIP_NO_ERROR)
+            if (err != 0)
```
Comment (bzbarsky-apple) [inline src/platform/ESP32/nimble/BLEManagerImpl.cpp:473]: This change doesn't really make sense.  If the point is that the value returned by `ble_gap_adv_stop` is not a `CHIP_ERROR`.... well, we're assigning it to a CHIP_ERROR.  Maybe we should have a separate variable for the non-CHIP-error return values in this function?

Doing that in #4256 is fine.

**Review by bzbarsky-apple — APPROVED** [review]

File: src/platform/ESP32/nimble/BLEManagerImpl.cpp (hunk 2 above)
Comment (dhrishi) [reply src/platform/ESP32/nimble/BLEManagerImpl.cpp:1051]: A call was made to `StartAdvertising` with updated advertisement data, when the non-connectable advertising was active and it returned error from this function (as the `if` check had the connectable flag). As a result, in `DriveBLEState` the service mode was set to `kCHIPoBLEServiceMode_Disabled` and advertisement was stopped.

File: src/platform/ESP32/nimble/BLEManagerImpl.cpp (hunk 3 above)
Comment (dhrishi) [reply src/platform/ESP32/nimble/BLEManagerImpl.cpp:473]: Ohh, my bad. I totally missed doing that. I have updated the PR with the change. Thanks!

Comment (dhrishi) [reply src/platform/ESP32/nimble/BLEManagerImpl.cpp]: Ohh. I totally missed doing that. I have updated the PR with the changes. Also, made similar change in `StartAdvertising` to not use `CHIP_ERROR` type variable for ESP32 NimBLE APIs. PTAL

**Review by mspang — APPROVED** [review]

**Comment by dhrishi** [comment]
Can someone merge this, if it looks okay?

**Pull request merged**